contrib/IECoreArnold/src/IECoreArnold/bindings/UniverseBlockBinding.cpp

```diff
@@ -39,6 +39,30 @@
 using namespace boost::python;
 using namespace IECoreArnold;
 
+namespace
+{
+
+object universeWrapper( UniverseBlock &universeBlock )
+{
+	AtUniverse *universe = universeBlock.universe();
+	if( universe )
+	{
+		object arnold = import( "arnold" );
+		object ctypes = import( "ctypes" );
+		return ctypes.attr( "cast" )(
+			(uint64_t)universeBlock.universe(),
+			ctypes.attr( "POINTER" )( object( arnold.attr( "AtUniverse" ) ) )
+		);
+	}
+	else
+	{
+		// Default universe, represented as `None` in Python.
```

Comment:
This implies that the default universe would consistently be represented as `None` in Python, but this is not true when IECOREARNOLD_MULTIPLE_UNIVERSES is set.  Do we want a principle that anywhere we take an AtUniverse should accept None or nullptr to use the default universe, and if so, should universe return nullptr for the default universe instead of returning g_apiScope.sharedUniverse()?

Reply:
"Default" here means "Arnold's default universe, the one implicitly created by `AiBegin()`". This is consistently represented by `nullptr` in the Arnold API (the deprecated APIs are implemented by passing `nullptr` to the non-deprecated APIs), and `None` in the Arnold Python API. It's true that we don't use it when `IECOREARNOLD_MULTIPLE_UNIVERSES` is set, but that doesn't mean that we don't want to translate `nullptr` to `None` unconditionally when returning a universe to Python. That branch won't actually be hit when `IECOREARNOLD_MULTIPLE_UNIVERSES` is set, but the logic is still valid, and suitable for anywhere we want to pass a universe from C++ to Python.

> Do we want a principle that anywhere we take an AtUniverse should accept None or nullptr to use the default universe

Yes, that's what I've aimed for here. The `pythonObjectToAtUniverse()` function in `NodeAlgoBinding.cpp` is the other half of this, implementing the `None -> nullptr` conversion when passing a universe from Python to C++.

Reply:
Does this make sense @danieldresser-ie? Are we OK to merge?

Reply:
Oh, sorry for the delay.  Yep, that seems fine, good to merge.